Sources/Target/Linux/X86/ProcessX86.cpp

```diff
@@ -22,54 +22,44 @@ namespace Linux {
 
 ErrorCode Process::allocateMemory(size_t size, uint32_t protection,
                                   uint64_t *address) {
-  if (address == nullptr)
+  if (address == nullptr) {
     return kErrorInvalidArgument;
-
-  ProcessInfo info;
-  ErrorCode error = getInfo(info);
-  if (error != kSuccess)
-    return error;
+  }
 
   U8Vector codestr;
   X86Sys::PrepareMmapCode(size, protection, codestr);
 
-  //
-  // Code inject and execute
-  //
-  error = ptrace().execute(_currentThread->tid(), info, &codestr[0],
-                           codestr.size(), *address);
-  if (error != kSuccess)
+  uint64_t result;
+  ErrorCode error = executeCode(codestr, result);
```

Comment:
`executeCode` takes a reference.

On Tue, May 10, 2016 at 2:44 AM, kokarez <notifications@github.com> wrote:

> In Sources/Target/Linux/X86/ProcessX86.cpp
> <https://github.com/facebook/ds2/pull/335#discussion_r62641395>:
>
> >
> >    U8Vector codestr;
> >    X86Sys::PrepareMmapCode(size, protection, codestr);
> >
> > -  //
> > -  // Code inject and execute
> > -  //
> > -  error = ptrace().execute(_currentThread->tid(), info, &codestr[0],
> > -                           codestr.size(), *address);
> > -  if (error != kSuccess)
> > +  uint64_t result;
> > +  ErrorCode error = executeCode(codestr, result);
>
> &result ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/335/files/ab7faaffbb4896bfd174d24c541b189bc519fcd5..e4293dcaed032942ef8b6def7335357527b5d108#r62641395>
>



-- 
Stephane Sezer